### PR TITLE
WAI-816 export bugs fixes

### PR DESCRIPTION
**Pull request by tcaiger**
### Issue #: WAI-816: export bugs fixes

Relates to:
- WAI-817: Bar Chart and Legend Overlap
- WAI-818: Bad Rounding in Chart
- WAI-879: Top Part of Numbers Cut Off
- WAI-1032: Axis labels aren't included in exports
- WAI-1043: Date of Data in PNG export from LESMIS is missing
- PHX-116: Matrix Table Export to include Date Selected
- WAI-1041: Can’t see numbers on stacked charts & long labels

### CODE REVIEW

File: packages/ui-components/src/components/Chart/CartesianChart.js [hunk 1]
```diff
@@ -160,7 +167,7 @@ export const CartesianChart = ({ viewContent, isEnlarged, isExporting, legendPos
     Object.keys(chartConfig).length > 0 ? chartConfig : { [DEFAULT_DATA_KEY]: {} };
   const hasLegend = hasDataSeries || renderLegendForOneItem;
   const aspect = !isEnlarged && !isMobile() && !isExporting ? 1.6 : undefined;
-  const height = isExporting || (isEnlarged && hasLegend && isMobile()) ? 320 : undefined;
+  const height = getHeight(isExporting, isEnlarged, hasLegend);
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/CartesianChart.js:170]: Ugh yeah great move clarifying this horrendous ternary!

File: packages/ui-components/src/components/Chart/PieChart.js [hunk 2]
```diff
@@ -80,27 +87,51 @@ const makeCustomTooltip = ({ valueType, labelType }) => {
...
+  };
+};
+
 const chartColorAtIndex = (colorArray, index) => {
   return colorArray[index % colorArray.length];
 };
 
+const getHeight = (isExporting, isEnlarged) => {
+  if (isExporting) {
+    return 420;
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/PieChart.js:117]: Yeeeh boi!

File: packages/ui-components/src/components/Chart/XAxis.js [hunk 3]
```diff
@@ -32,12 +32,28 @@ const X_AXIS_PADDING = {
...
 
-export const XAxis = ({ viewContent, isExporting, isEnlarged }) => {
-  const [xAxisHeight, setXAxisHeight] = useState(0);
-  const fillColor = getContrastTextColor();
+const renderXAxisLabel = (label, fillColor, isEnlarged, isExporting) => {
+  if (label)
+    return {
+      value: label,
+      fill: fillColor,
+      position: isExporting ? 'bottom' : 'bottom',
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/XAxis.js:40]: No different?

File: packages/ui-components/src/components/Chart/XAxis.js [hunk 4]
```diff
@@ -32,12 +32,28 @@ const X_AXIS_PADDING = {
...
+
+const calculateXAxisHeight = data => {
+  return Math.min(BASE_H + Math.max(...data.map(item => item.name.length)) * 6, 190);
+};
+
+export const XAxis = ({ viewContent, isExporting, isEnlarged }) => {
+  const fillColor = isExporting ? DARK_BLUE : getContrastTextColor();
   const { BAR, COMPOSED } = CHART_TYPES;
   const { chartType, chartConfig = {}, data } = viewContent;
+  const axisHeight = isExporting ? calculateXAxisHeight(data) : BASE_H;
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/XAxis.js:56]: I would move the ternary into the function personally

File: packages/ui-components/src/components/Chart/PieChart.js [hunk 5]
```diff
@@ -80,27 +87,51 @@ const makeCustomTooltip = ({ valueType, labelType }) => {
...
+};
+
 export const PieChart = ({ viewContent, isExporting, isEnlarged, onItemClick, legendPosition }) => {
   const [activeIndex, setActiveIndex] = useState(-1);
+  const { presentationOptions, data } = viewContent;
+  // eslint-disable-next-line no-unused-vars
+  const [loaded, setLoaded] = useState(false);
+
+  // Delay rendering of the chart to fix an issue with the legend overlapping the chart.
+  // This is a work around for a recharts bug. @see https://github.com/recharts/recharts/issues/511
```
Comment (edmofro) [inline packages/ui-components/src/components/Chart/PieChart.js:129]: But is it being used? I can't see `loaded` referenced anywhere (haven't checked out locally though)

**Review by edmofro — CHANGES_REQUESTED** [review]
So many fixes!!!

File: packages/ui-components/src/components/Chart/PieChart.js (hunk 5 above)
Comment (tcaiger) [reply packages/ui-components/src/components/Chart/PieChart.js:129]: I've updated the comment to be a bit more accurate but all this does is triggers a re-render of the component and that does the trick. This change actually belongs to the WAI-2017 ticket but I've got it here and as I'm anticipating WAI-1027 getting into dev well before these changes.

Comment (edmofro) [reply packages/ui-components/src/components/Chart/PieChart.js:129]: Aah right that makes sense!

**Review by edmofro — APPROVED** [review]
lgtm 😃

**Pull request merged**